.github/workflows/main.yml

```diff
@@ -0,0 +1,58 @@
+name: Main workflow
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          #- macos-latest
+          #- windows-latest
+        ocaml-version:
+          - 4.09.1
+          - 4.08.1
+          - 4.07.1
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      - name: Pinning BAP
+        run: opam pin add bap . --no-action
+
+      - name: Configuring Homebrew LLVM
+        if: matrix.os == 'macos-latest'
+        run: echo '::set-env name=LLVM_CONFIG::/usr/local/opt/llvm/bin/llvm-config'
+
+      - name: Installing system dependencies
+        run: opam depext bap --yes --with-doc --with-test
+
+      - name: Installing opam dependencies
+        run: opam install . --deps-only --with-doc --with-test
+
+      - name: Install BAP
+        run: opam install bap -v --yes --with-test
+
+      - name: running the testsuite
+        run: |
+          eval $(opam env)
+          cd testsuite
+          git submodule init
```

Comment:
I think those unnecessary since we already cloned with submodules recursively.

Reply:
well, that's one of the problems - I don't see that it's done recursively :(

Reply:
even if `submodule=true` won't work (which I doubt) it is still not needed, as our `Makefile` is already doing this. Thus doing, `make check` is enough even if submodules are not initialized. 

So the question, why just not?
```
name: Running unit tests
run: make test

name: Running functional tests
run: make check
```